main.go

```diff
@@ -110,6 +120,39 @@ func main() {
 	<-ctx.Done()
 }
 
+func startDevicePluginServer(ctx context.Context, config *Config) error {
+	manager := deviceplugin.NewManager(config.DevicePluginPath)
+
+	server := deviceplugin.NewServer(config.Name, config.ResourceCount, config.HostBaseDir, config.HostPathEnv)
+	if err := server.Start(ctx, manager); err != nil {
+		return err
+	}
+
+	kubeletMonitorCh, err := manager.MonitorKubeletRestart(ctx)
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case _, ok := <-kubeletMonitorCh:
+				if !ok {
+					return
+				}
+			}
+			_ = server.Stop()
```

Comment:
I think it will be enough to just re-register to Kubelet, without need to stop our grpc server on unix socket.

Reply:
Fixed